src/core/ddsi/src/ddsi__protocol.h

```diff
@@ -264,6 +264,7 @@ typedef union ddsi_rtps_submessage {
 #define DDSI_PID_TRANSPORT_PRIORITY                  0x49u
 #define DDSI_PID_PROTOCOL_VERSION                    0x15u
 #define DDSI_PID_VENDORID                            0x16u
+#define DDSI_PID_PRODUCT_VERSION                     0x8000u
```

Comment:
The vendor-specific flags are defined below, e.g., line 349 where it defines `DDSI_PID_ADLINK_PARTICIPANT_VERSION_INFO`. 0x8000 is already defined in this file, inherited from OpenSplice but not used by Cyclone (see https://github.com/eclipse-cyclonedds/cyclonedds/blob/d8cef891b9a27cc7bc52477e1c7b2c3f8edf81ba/src/core/ddsi/src/ddsi_plist.c#L2110 and the table below it: it doesn't occur in either). There are a lot of parameters there that it doesn't grok, come to think of it.

I suspect `0x8000` has never been used by Cyclone, so I think it can be reclaimed.